### ISSUE TITLE
integration.modules.test_shadow.ShadowModuleTest failures under py2 on fedora28

### ISSUE THREAD

**Issue by waynew** [saltstack/salt#51926]
Two failures that are probably related:

- [ ] https://jenkinsci.saltstack.com/job/2017.7/job/salt-fedora-28-py2/52/testReport/junit/integration.modules.test_shadow/ShadowModuleTest/test_set_expire/
- [ ] https://jenkinsci.saltstack.com/job/2017.7/job/salt-fedora-28-py2/52/testReport/junit/integration.modules.test_shadow/ShadowModuleTest/test_set_date/

```
Traceback (most recent call last):
  File "/tmp/kitchen/testing/tests/support/helpers.py", line 123, in wrap
    return caller(cls)
  File "/tmp/kitchen/testing/tests/integration/modules/test_shadow.py", line 195, in test_set_date
    self.assertTrue(self.run_function('shadow.set_date', [self._test_user, '2016-08-19']))
AssertionError: False is not true
```

**Comment by waynew** [comment]
I'm not sure if the fedora29 build failures are related to the whole shadow thing - see https://jenkinsci.saltstack.com/job/2017.7/job/salt-fedora-29-py2/42/console

```
04:38:36        unittest.case.SkipTest: WARNING(SHOULD NOT HAPPEN #1935): Failed to get 'shadow.gen_password' from the minion 'minion'. Command output: {'minion': None}
04:38:36        Copying /tmp/xml-unittests-output to artifacts/
04:38:37        chmod: cannot access '/tmp/coverage.xml': No such file or directory
04:38:37 >>>>>> ------Exception-------
04:38:37 >>>>>> Class: Kitchen::ActionFailed
04:38:37 >>>>>> Message: 1 actions failed.
04:38:37 >>>>>>     Failed to complete #verify action: [SSH exited (1) for command: [sudo -E chmod -R +r /tmp/coverage.xml]] on py2-fedora-29
```

If that's unrelated we should create a separate issue for it.

**Comment by dwoz** [comment]
This is also failing on develop
```
integration.modules.test_shadow.ShadowModuleTest.test_set_date
--------------------------------------------------------------------------------
develop failed salt-fedora-28-py2, salt-fedora-28-py3
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
2017.7 failed salt-fedora-28-py3
--------------------------------------------------------------------------------
```

**Comment by Ch3LL** [comment]
i believe this might fix the failures on 2017.7 atleast: https://github.com/saltstack/salt/pull/52155

as that was previously failing on 2018.3

**Comment by Ch3LL** [comment]
verified is passing on 2017.7.9
